viewing_party/party.py

```diff
@@ -1,23 +1,160 @@
 # ------------- WAVE 1 --------------------
-
+import copy
+#Creates a dictionary representing a movie with title, genre and rating.
 def create_movie(title, genre, rating):
-    pass
-
+    if title and genre and rating:
+        return {"title": title, "genre": genre, "rating": rating} #it's a movie_dic
+    return None
 # -----------------------------------------
-# ------------- WAVE 2 --------------------
+#Adds a movie to the user's watched list.
+def add_to_watched(user_data, movie): 
+    if "watched" not in user_data:
+        user_data["watched"] = []
+    user_data["watched"].append(movie)
+    return user_data
 # -----------------------------------------
+#Adds a movie to the user's watchils.
+def add_to_watchlist(user_data, movie):
+    if "watchlist" not in user_data:
+        user_data["watchlist"] = []
+    user_data["watchlist"].append(movie)
+    return user_data
+# -----------------------------------------
+#Moves a movie from the watchlist to the watched list.
+def watch_movie(user_data, title):
+    #watchlist_copy = user_data["watchlist"][:] #shallow copy syntax
+    user_data_copy = copy.deepcopy(user_data) # deep copy
+
+    for movie in user_data_copy["watchlist"]:
+        if movie["title"] == title:
+            user_data_copy["watchlist"].remove(movie)
+            user_data_copy["watched"].append(movie)
+            
+    return user_data_copy
+# ------------- WAVE 2 --------------------
+#Calculates the average rating of all watched movies.
+def get_watched_avg_rating(user_data):
+    if "watched" not in user_data or len(user_data["watched"]) == 0:

```

Comment:
This is a nice guard clause to check for invalid input and immediately return 0.0. This allows the main logic to be unindented and not be inside a conditional block. 

A more Pythonic way to write the expression to check if the `watched` list is not empty would be `not user_data["watched"]`
```suggestion
    if "watched" not in user_data or not user_data["watched"]:
```